File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOSupportMaxSetItemsNumModel.kt
//
//  IDOSupportMaxSetItemsNumModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Get maximum number of settings supported by firmware event number

open class IDOSupportMaxSetItemsNumModel(
    contactMaxSetNum: Int,
    reminderMaxSetNum: Int,
    msgMaxBuffSize: Int
) : IDOBaseModel {

    /// Maximum number of frequently contacted persons that firmware supports for app to set (default is 10)
    @SerializedName("contact_max_set_num")
    var contactMaxSetNum: Int = contactMaxSetNum

    /// Maximum number of schedule reminders that firmware supports for app to set(default is 30)
    @SerializedName("reminder_max_set_num")
    var reminderMaxSetNum: Int = reminderMaxSetNum

    /// Maximum sending buffer size of message reminders (default is 250 bytes)
    @SerializedName("msg_max_buff_size")
    var msgMaxBuffSize: Int = msgMaxBuffSize


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOSupportMaxSetItemsNumModel.swift
//
//  IDOSupportMaxSetItemsNumModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Get maximum number of settings supported by firmware event number
@objcMembers
public class IDOSupportMaxSetItemsNumModel: NSObject, IDOBaseModel {
    /// Maximum number of frequently contacted persons that firmware supports for app to set (default is 10)
    public var contactMaxSetNum: Int
    /// Maximum number of schedule reminders that firmware supports for app to set(default is 30)
    public var reminderMaxSetNum: Int
    /// Maximum sending buffer size of message reminders (default is 250 bytes)
    public var msgMaxBuffSize: Int

    enum CodingKeys: String, CodingKey {
        case contactMaxSetNum = "contact_max_set_num"
        case reminderMaxSetNum = "reminder_max_set_num"
        case msgMaxBuffSize = "msg_max_buff_size"
    }

    public init(contactMaxSetNum: Int, reminderMaxSetNum: Int, msgMaxBuffSize: Int) {
        self.contactMaxSetNum = contactMaxSetNum
        self.reminderMaxSetNum = reminderMaxSetNum
        self.msgMaxBuffSize = msgMaxBuffSize
    }

    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}
